test-data/unit/check-overloading.test

```diff
@@ -6456,7 +6456,7 @@ P = ParamSpec("P")
 R = TypeVar("R")
 
 @overload
-def func(x: Callable[Concatenate[Any, P], R]) -> Callable[P, R]: ...  # E: Overloaded function signatures 1 and 2 overlap with incompatible return types
```

Comment:
I forgot to check why this error disappeared. I will take a look at it if you think it is important.

Reply:
I don't know how important this is, though the error is correct. Maybe (I haven't looked through the code yet) you're saying the `P` here is the same as the `P` in the other overload?

Reply:
Looking at the `mypy_primer` output, I see there is something suspicious going on with overloads, so I will dig a bit deeper into this.

Reply:
OK, I fixed one of the new overload errors in `mypy_primer`, this one however looks quite tricky to bring back. But note that when I enable `--new-type-inference` (also during unification manually, that flag doesn't control it) the error re-appears. So I guess we can just wait until it will be on by default.

Reply:
Yeah, this doesn't look like a serious regression.